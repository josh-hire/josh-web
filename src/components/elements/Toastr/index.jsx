import React, { useContext } from 'react';
import { ToastrContext } from '../../../contexts/ToastrContext';
import { IconCircleCheck } from '../Logo/circleCheck';
import { IconClose } from '../Logo/close';
import { IconSad } from '../Logo/sad';
import styles from './styles.module.scss';
import ReactDom from 'react-dom';

const Toastr = () => {
  const { toastr, setToastr } = useContext(ToastrContext);

  const initialColor = [
    { key: 'success', style: styles.green, icon: IconCircleCheck },
    { key: 'error', style: styles.red, icon: IconSad },
  ];
  const toastStyles = initialColor.find((val) => val.key === toastr.color);
  const Icon = toastStyles?.icon;

  const handleCloseToastr = () => {
    setToastr({
      message: '',
      color: '',
      isOpen: false,
    });
  };

  const renderToastr = () => {
    return ReactDom.createPortal(
      <div className={[styles.container, toastStyles?.style].filter(Boolean).join(' ')}>
        <Icon />
        <p>{toastr.message || 'Network Error'}</p>
        <button onClick={() => handleCloseToastr()}>
          <IconClose />
        </button>
      </div>,
      document.getElementById('toastr'),
    );
  };

  return (
    <>
      <div id="toastr" />
      {toastr.isOpen && renderToastr()}
    </>
  );
};

export default Toastr;
