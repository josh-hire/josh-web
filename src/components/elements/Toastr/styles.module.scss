@import 'Styles/utilities/variables';

.container {
  width: auto;
  position: fixed;
  left: 50%;
  transform: translateX(-50%);
  top: 1rem;
  border-radius: 1rem;
  display: flex;
  align-items: center;
  padding: .5rem .75rem;
  gap: 1rem;
  z-index: 1000;

  p {
    margin: 0;
    font-size: 1rem;
  }

  button {
    background-color: unset;
    border: none;
    cursor: pointer;
    display: flex;
  }

  &.green {
    background-color: $color-success-10;
    color: #25282B;

    svg {
      fill: $color-success;
    }
  }

  &.red {
    background-color: $color-error-10;
    color: #25282B;

    svg {
      fill: $color-error;
    }
  }
}