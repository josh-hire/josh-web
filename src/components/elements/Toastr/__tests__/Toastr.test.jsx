import '@testing-library/jest-dom';
import Toastr from '../index';
import ShallowRenderer from 'react-test-renderer/shallow';
import '@testing-library/jest-dom';
import ContextProvider from 'Contexts/index';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState:jest.fn(),
  useContext:()=>({ toastr:{ color:'red', isOpen:true }, setToastr:jest.fn() }),
}));

jest.mock('react-dom',()=>({
  ...jest.requireActual('react-dom'),
  createPortal:(a)=>a,
  render: jest.fn(),
}));

describe('src/component/elements/Toastr', ()=>{
  test('Toastr render',()=>{
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ContextProvider><Toastr /></ContextProvider>);
    expect(tree).toMatchSnapshot();
  });

  test('renderToastr',()=>{
    const shallow = new ShallowRenderer();
    const tree = shallow.render(
      <Toastr isOpen={true} />
    );
    const treeWrap =tree.props.children[1];
    treeWrap.props.children[2].props.onClick();
  });
});
