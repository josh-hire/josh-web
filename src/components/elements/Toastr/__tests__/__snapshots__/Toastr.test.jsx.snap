// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/component/elements/Toastr Toastr render 1`] = `
<ReactQueryContextProvider>
  <MantineContextProvider>
    <AuthContextProvider>
      <ToastrContextProvider>
        <Toastr />
      </ToastrContextProvider>
    </AuthContextProvider>
  </MantineContextProvider>
</ReactQueryContextProvider>
`;
